web/src/lib/components/elements/dropdown.svelte

```diff
@@ -18,13 +23,18 @@
     value = options[index];
     showMenu = false;
   };
+
+  $: index = options.findIndex(option => option === value)
+  $: icon = icons?.[index]
 </script>
 
 <div id="dropdown-button" use:clickOutside on:outclick={handleClickOutside}>
   <!-- BUTTON TITLE -->
   <LinkButton on:click={() => (showMenu = true)}>
     <div class="flex place-items-center gap-2 text-sm">
-      <SwapVertical size="18" />
+      {#if icon}
```

Comment:
This should be `value.icon` and you should not need lines 26-28 or 14 anymore.

Reply:
Basically the interface is now "pass in a list of icon+label". No need to pass in two separate lists now. When the value is selected, it's associated icon (if any) will show up here.

Reply:
A _issue_ with this would be that the store value we bind to from the dropdown would now also need to expect a object of icon+label and it's interface would need to look like:

```js
export interface AlbumViewSettings {
  sortBy: {
    value: string;
    icon?: typeof LinkButton
  };
  view: {
    value: string;
    icon?: typeof LinkButton
  };
}
```

Usually I would solve this by having a separate interface file e.g. `interfaces/IDropdownOption.ts` and then import it in `Dropdown.svelte` and `preferences.store.ts`, but I do not see a similar project structure. 
Do you have a suggestion on how to solve this cleanest possible?

Reply:
Also I wonder if the interface might want to look like:

```js
interface DropdownOption {
  name: string;
  icon?: typeof LinkButton;
}
```

so we references to the text label would be `value.name` instead of `value.value`.

Reply:
Yeah, that looks better. I didn't realize this was persisted. It does make it a bit more complicated.